dashboard/src/components/Filter/CodeBlock.tsx

```diff
@@ -21,119 +33,204 @@ interface IHighlightedCode {
+          <ColoredCircle
+            quantity={highlightedCode.failCount}
+            backgroundClassName="bg-lightRed"
+          />
+          <FormattedMessage id="global.fails" defaultMessage={'Fails'} />
+        </li>
+        <li className="flex gap-1">
+          <ColoredCircle
+            quantity={highlightedCode.errorCount}
+            backgroundClassName="bg-orange-200"
+          />
+          <FormattedMessage id="global.errors" defaultMessage={'Errors'} />
+        </li>
+        <li className="flex gap-1">
+          <ColoredCircle
+            quantity={
+              highlightedCode.highlightCount -
+              highlightedCode.failCount -
+              highlightedCode.errorCount
+            }
+            backgroundClassName="bg-mediumGray"
+          />
+          <FormattedMessage id="global.others" defaultMessage={'Others'} />
+        </li>
+      </ul>
+      <div className="flex items-center">
+        <LiaInfoCircleSolid />
+        <p className="pl-1 text-sm">
+          <FormattedMessage
+            id="codeBlock.highlightsTooltip"
+            defaultMessage="Test"
+          />
+        </p>
+      </div>
+    </div>
+  );
+};
+
+const generateHighlightedCode = (code: string): IHighlightedCode => {
+  let highlights = 0;
+  let fails = 0;
+  let errors = 0;
+
+  // matches any < or > and replaces with equivalent &lt or &gt
+  let newCode = code.replace(/[<>]/gm, match => {
+    match = match.replace(/</, '&lt');
+    match = match.replace(/>/, '&gt');
+    return match;
+  });
+
+  newCode = newCode.replace(
+    // matches any line with the occurrence of error or fail
+    /^.*(error|fail).*$/gim,
+    match => {
+      highlights++;
+      if (
+        // matches failed to/with, more than 0 fails/failed and no flags
+        match.search(
+          /.*((\bfailed(\s*to|\s*with|([\b\s:]\s*\(*-*[1-9])))|fail(\b|:|\s*[1-9])(?![|/]))/i,
+        ) !== -1
+      ) {
+        fails++;
+        return '<span class="text-red">' + match + '</span>';
+      }
+      // matches error codes greater than 0 or more than 0 errors
+      if (
+        match.search(
+          /(error(:|,|[\b\s:]\s*\(*-*[1-9a-z]))|([1-9]\s*error)/i,
+        ) !== -1
+      ) {
+        errors++;
+        return '<span class="text-orange-500">' + match + '</span>';
+      }
+      return '<span class="text-sky-600">' + match + '</span>';
+    },
+  );
+  return {
+    highlightedCode: newCode,
+    highlightCount: highlights,
+    failCount: fails,
+    errorCount: errors,
+  };
+};
+
+const CodeBlock = ({
+  code,
+  highlightsClassnames,
+}: TCodeBlockProps): JSX.Element => {
+  const highlightedCode: IHighlightedCode = useMemo(
+    () => generateHighlightedCode(code),
+    [code],
+  );
+
+  const footerElement =
+    highlightedCode.highlightCount > 0 ? (
+      <HighlightCounts
+        highlightedCode={highlightedCode}
+        highlightsClassnames={highlightsClassnames}
+      />
+    ) : undefined;
```

Comment:
You don't need undefined here,
```suggestion
  const footerElement =
    highlightedCode.highlightCount > 0 && (
      <HighlightCounts
        highlightedCode={highlightedCode}
        highlightsClassnames={highlightsClassnames}
      />
    );
```